### PR TITLE
feat: Add `load_balancers` parameter to associate a CLB (Classic Load Balancer) to worker groups ASG

### PR DESCRIPTION
**Pull request by hacker65536**
# PR o'clock

## Description

Add `load_balancers`  for clb be associated to asg.

Resolves https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1036

### Checklist

- [ ] CI tests are passing
- [ ] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by dpiddockcmp** [comment]
Please can you duplicate this change in the ASG in `workers_launch_template.tf`, thank you.

**Comment by barryib** [comment]
@hacker65536 Thanks for your contribution.

I was just wondering what is your actual use case. What you're trying to achieve by adding a load balancer to all your worker nodes in a kubernetes cluster. Is the kubernetes model does't fit here (using k8s service or ingress) ?

**Comment by hacker65536** [comment]
I'm trying to using clb + envoy to handle grpc app.

File: workers_launch_template.tf [hunk 1]
```diff
@@ -37,6 +37,11 @@ resource "aws_autoscaling_group" "workers_launch_template" {
     "target_group_arns",
     local.workers_group_defaults["target_group_arns"]
   )
+  load_balancers = lookup(
+    var.worker_groups[count.index],
```
[the remark this answers is not in the document]
Comment (oraclef2k20) [reply workers_launch_template.tf:41]: Thank you

**Review by barryib — APPROVED** [review]

**Comment by barryib** [comment]
Thanks @hacker65536 for your contribution.

**Comment by mumoshu** [comment]
> Is the kubernetes model does't fit here (using k8s service or ingress) ?

Using K8s service LBs makes it harder to recreate the cluster, because you need to use e.g. Route 53 to switch traffic from the old to the new cluster. So I'd use service LBs and (current) AWS ingresses only for dev clusters that can be deleted without care. So this feature should generally be useful for any production setup. Just my two cents 😃

**Comment by barryib** [comment]
> > Is the kubernetes model does't fit here (using k8s service or ingress) ?
> 
> Using K8s service LBs makes it harder to recreate the cluster, because you need to use e.g. Route 53 to switch traffic from the old to the new cluster. So I'd use service LBs and (current) AWS ingresses only for dev clusters that can be deleted without care. So this feature should generally be useful for any production setup. Just my two cents 😃

Great. Thanks @mumoshu for the clarification.

**Pull request merged**